clients/src/test/java/org/apache/kafka/common/config/ConfigDefTest.java

```diff
@@ -164,6 +170,155 @@ public void testNullDefaultWithValidator() {
         assertEquals("ONE", vals.get(key));
     }
 
+    @Test
+    public void testParseNoCheckRequired() {
+        ConfigDef def = new ConfigDef()
+            .define("a", Type.INT, Importance.HIGH, "docs")
+            .define("b", Type.INT, Importance.HIGH, "docs");
+
+        Map<String, String> props = new HashMap<>();
+        props.put("a", "1");
+
+        Map<String, Object> parsed = def.parse(props, false);
+        assertEquals(1, (int) parsed.get("a"));
+    }
+
+    @Test(expected = ConfigException.class)
+    public void testCannotParse() {
+        ConfigDef def = new ConfigDef()
+            .define("a", Type.INT, Importance.HIGH, "docs")
+            .define("b", Type.INT, Importance.HIGH, "docs");
+
+        Map<String, String> props = new HashMap<>();
+        props.put("b", "non_integer");
+        def.parse(props, false);
+    }
+
+    @Test
+    public void testGroupInference() {
+        List<String> expected = Arrays.asList("group1", "group2");
+        ConfigDef def = new ConfigDef()
+            .define("a", Type.INT, Importance.HIGH, "docs", "group1", 1, Width.SHORT, "a")
+            .define("b", Type.INT, Importance.HIGH, "docs", "group2", 1, Width.SHORT, "b")
+            .define("c", Type.INT, Importance.HIGH, "docs", "group1", 2, Width.SHORT, "c");
+
+        List<String> groups = def.groups();
+        assertEquals(expected, groups);
```

Comment:
Since ordering matters for groups, you might want to test a different ordering of the same config definitions.